conda/recipes/ucx-py/meta.yaml

```diff
@@ -18,7 +18,7 @@ build:
   number: {{ GIT_DESCRIBE_NUMBER }}
   string: py{{ py_version }}_{{ date_string }}_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
   script:
-    - {{ PYTHON }} -m pip install . -vv
+    - {{ PYTHON }} -m pip install --no-build-isolation --no-deps . -vv
```

Comment:
Without this, conda builds tried to install `libucx` (no suffix) and failed complaining about that library not existing.

Using `--no-deps` prevents that. I think this is a better pattern *anyway*... it defers 100% of the responsibility for providing build dependencies to `conda-build`, which means we'll get conda packages of all those things and find out via build failures if some required `build` / `host` dependency was accidentally omitted from this recipe.

Reply:
My understanding was that `conda-build` automatically configured the environment not to install pip dependencies by setting the `PIP_NO_DEPENDENCIES` environment variable, see https://github.com/conda/conda-build/blob/main/conda_build/build.py#L3008. IOW I thought these two flags would be implied and superfluous. Let's double-check that with some of our conda experts to see why they might be necessary before we commit this.

Reply:
hmmm this *might* have been before I added the `if not r.startswith("libucx")` a few lines down below, in the part where we read in all the host + run dependencies from the corresponding tables in `pyproject.toml`.

I can doble-check that.

Reply:
```suggestion
    - {{ PYTHON }} -m pip install . -vv
```

Reply:
Agree this shouldn't be needed, but the additional flags are safe to add

Reply:
Yep looks like that was the problem... filtering `libucx` out of every place where we read `pyproject.toml` deps into conda recipe deps with jinja was enough, and these flags were unnecessary.

successful conda builds: https://github.com/rapidsai/ucx-py/actions/runs/8991800910/job/24700209674?pr=1041

Thanks for noting it, I didn't know before this conversation that `--no-deps` was the default in `conda-build`, that's helpful!

Reply:
Thanks @jakirkham , looks like we posted at the same time. As long as it's not needed, I'll leave this reverted to shrink the size of the changeset here.